gusto/linear_solvers.py

```diff
@@ -322,13 +134,13 @@ def __init__(self, state, quadrature_degree=None, solver_parameters=None,
 
         if logger.isEnabledFor(DEBUG):
             # Set outer solver to GMRES and turn on KSP monitor for the outer system
-            self.solver_parameters["ksp_type"] = "gmres"
+            self.solver_parameters["ksp_type"] = "fgmres"
             self.solver_parameters["mat_type"] = "aij"
             self.solver_parameters["pmat_type"] = "matfree"
-            self.solver_parameters["ksp_monitor_true_residual"] = True
+            self.solver_parameters["ksp_monitor_true_residual"] = None
 
             # Turn monitor on for the trace system
-            self.solver_parameters["condensed_field"]["ksp_monitor_true_residual"] = True
+            self.solver_parameters["condensed_field"]["ksp_monitor_true_residual"] = None
```

Comment:
I don't fully understand these changes, but happy to trust you on them. Only thing I was wondering is whether the comment on line 142 still matches what you are doing in 143?

Reply:
Ah, this is because of a recent options change made in Firedrake. I don't really like it either, but basically booleans like `True` and `False` are no longer accepted when you want to turn on certain PETSc options. Options that are normally called via the command line act like switches, such as `python foo.py --options_prefix_ksp_monitor` would turn on the monitor. You don't pass `True`, you just pass the command. So now we just pass `None` so something is there. Then it gets easier to translate this when we pass options to PETSc.

Reply:
Ah cool, thanks for the explanation!